packages/shared/src/components/MainLayout.tsx

```diff
@@ -153,6 +154,20 @@ function MainLayoutComponent({
     router.push(`${onboardingUrl}?${params.toString()}`);
   }, [shouldRedirectOnboarding, router]);
 
+  const utmSource = router?.query?.utm_source;
```

Comment:
is this a param coming from both notifications in app, push notifications and emails?

Reply:
Yeah, if it is in-app, that means they are logged in. For both email and push notifications, they both have this param.